File: TypeScript/codewars/Detect-Pangram.ts
export const isPangram = (phrase: string): boolean => {
	const regex = /[^a-zA-Z]/g;

	const setPhraseArray: Set<string> = new Set<string>(
		phrase
			.replace(regex, "")
			.split("")
			.map((value) => value.toLocaleLowerCase()),
	);

	return Array.from(setPhraseArray).length === 26;
};

File: Csharp/CodeWars/src/Completed/DetectPangram.cs
using System.Text.RegularExpressions;

namespace CodeWars.Completed;

public class DetectPangram
{
    public static bool IsPangram(string str) =>
        Regex.Replace(str.ToLower(), "[^a-z]", string.Empty).Distinct().Count() == 26;
}
